src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -530,7 +546,7 @@ private static ImmutableArray<Diagnostic> FilterDiagnosticsSuppressedInSource(Im
                     var workerTasks = new Task<CompilationCompletedEvent>[workerCount];
                     for (int i = 0; i < workerCount; i++)
                     {
-                        workerTasks[i] = ProcessCompilationEventsCoreAsync(analysisScope, analysisStateOpt, prePopulatedEventQueue, cancellationToken);
+                        workerTasks[i] = Task.Run(async() => await ProcessCompilationEventsCoreAsync(analysisScope, analysisStateOpt, prePopulatedEventQueue, cancellationToken).ConfigureAwait(false));
```

Comment:
`ProcessCompilationEventsCoreAsync` will callback into the analyzer on the same thread, and if that never returns we will never spawn other workers. This change ensures that all callbacks happen on background tasks and we always create all workers upfront.